app/controllers/api/v1/products_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  module V1
+    class ProductsController < ApplicationController
+      before_action :set_product, except: %i[create index]
+      def index
+        render json: { data: ActiveModel::SerializableResource.new(Product.all, each_serializer: ProductSerializer) }
+      end
+
+      def show
+        render json: ProductSerializer.new(@product).to_h
```

Comment:
Same here
```suggestion
        render json: @product
```
https://github.com/rails-api/active_model_serializers/blob/0-10-stable/docs/general/getting_started.md#rails-integration